pinot-core/src/main/java/org/apache/pinot/core/plan/maker/InstancePlanMakerImplV2.java

```diff
@@ -198,8 +198,13 @@ private int getMaxExecutionThreads(QueryContext queryContext) {
     Map<String, String> queryOptions = queryContext.getQueryOptions();
     if (queryOptions != null) {
       Integer maxExecutionThreads = QueryOptions.getMaxExecutionThreads(queryOptions);
```

Comment:
nit: can we rename `maxExecutionThreads` to sth like `maxExecutionThreadsFromQuery`?